src/Shared/ErrorUtilities.cs

```diff
@@ -736,12 +740,19 @@ internal static void VerifyThrowArgumentLength<T>(IReadOnlyCollection<T> paramet
         /// <param name="parameterName"></param>
         internal static void VerifyThrowArgumentLengthIfNotNull<T>(IReadOnlyCollection<T> parameter, string parameterName)
         {
-            if (parameter?.Count == 0 && s_throwExceptions)
+            if (parameter?.Count == 0)
             {
-                throw new ArgumentException(ResourceUtilities.FormatResourceStringStripCodeAndKeyword("Shared.ParameterCannotHaveZeroLength", parameterName));
+                ThrowArgumentLength(parameterName);
             }
         }
 #endif
+        private static void ThrowArgumentLength(string parameterName)
+        {
+            if (s_throwExceptions)
```

Comment:
@rainersigwald Can we safely get rid of s_throwExceptions ?
I believe there is no use for it any longer. It currently blocks some "nullable enable" refactorings as it does not recognize `VerifyThrowArgumentNull` as always throwing on given conditions.

Reply:
I don't know of a current use for it. My understanding is it was created as a super-duper-emergency escape hatch for customers to work around MSBuild bugs that cause exceptions but where the exception is actually benign. But that's super rare, I haven't used it in 10+ years, and we can ship fixes pretty quickly these days. File an issue and let's run it by the team but I'm ok with trying to remove it.

Reply:
I think it may date from my era? If so, do we recall ever needing to use it? If not then I think it would be safe to remove.